File: python/sqirvy_cli/sqirvy/llama_test.py
"""
Llama client test
"""

import os
from .llama_client import new_llama_client, Options

# Example Usage (optional, for testing)
if __name__ == "__main__":
    # Ensure LLAMA_API_KEY and LLAMA_BASE_URL are set
    if not os.getenv("LLAMA_API_KEY") or not os.getenv("LLAMA_BASE_URL"):
        print("Skipping example: LLAMA_API_KEY or LLAMA_BASE_URL not set.")
    else:
        try:
            TEST_MODEL = "llama3.3-70b"
            client = new_llama_client(TEST_MODEL)
            print("LlamaClient created successfully.")

            # Example query
            try:
                # Use default temp scale (2.0)
                opts = Options(temperature=1.0, max_tokens=100)
                # Ensure this model name matches what your Llama endpoint expects
                response = client.query_text(
                    "You are a helpful coding assistant.",
                    ["Explain the difference between a list and a tuple in Python."],
                    opts,
                )
                print(f"\nQuery Response from {TEST_MODEL}:")
                print(response)

                # Test empty prompt error
                print("\nTesting empty prompt error:")
                try:
                    client.query_text("System", [], opts)
                except ValueError as e:
                    print(f"Successfully caught expected ValueError: {e}")

            except ValueError as e:
                print(f"An unexpected error occurred during query: {e}")

            client.close()
            print("\nClient closed.")

        except ValueError as e:
            print(f"Configuration Error: {e}")

File: pkg/sqirvy/llama_test.go
package sqirvy

import (
	"context"
	"os"
	"testing"
)

func TestLlamaClient_QueryText(t *testing.T) {
	if os.Getenv("LLAMA_API_KEY") == "" {
		t.Skip("LLAMA_API_KEY not set")
	}
	if os.Getenv("LLAMA_BASE_URL") == "" {
		t.Skip("LLAMA_BASE_URL not set")
	}

	client, err := NewClient("llama")
	if err != nil {
		t.Fatalf("Failed to create client: %v", err)
	}

	tests := []struct {
		name    string
		prompt  []string
		wantErr bool
	}{
		{
			name:    "Basic prompt",
			prompt:  []string{"Say 'Hello, World!'"},
			wantErr: false,
		},
		{
			name:    "Empty prompt",
			prompt:  []string{},
			wantErr: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := client.QueryText(context.Background(), assistant, tt.prompt, "llama3.3-70b", Options{})
			if tt.wantErr {
				if err == nil {
					t.Errorf("LlamaClient.QueryText() error = %v, wantErr %v", err, tt.wantErr)
				}
				return
			}
			if err != nil {
				t.Errorf("LlamaClient.QueryText() error = %v", err)
				return
			}
			if len(got) == 0 {
				t.Error("LlamaClient.QueryText() returned empty response")
			}
		})
	}
}
